pkg/controller/admissionchecks/multikueue/jobset_adapter.go

```diff
@@ -62,9 +63,12 @@ func (b *jobsetAdapter) SyncJob(ctx context.Context, localClient client.Client,
 	if remoteJob.Labels == nil {
 		remoteJob.Labels = map[string]string{}
 	}
-	remoteJob.Labels[constants.PrebuiltWorkloadLabel] = workloadName
+	remoteJob.Labels[controllerconstants.PrebuiltWorkloadLabel] = workloadName
 	remoteJob.Labels[kueuealpha.MultiKueueOriginLabel] = origin
 
+	// set the manager
+	remoteJob.Labels[jobset.LabelManagedBy] = jobset.JobSetManager
```

Comment:
Without going more into code  I think we should first migrate JobSet to use managedBy field to be consistent with batch Job. Could you please send the JobSet PR first?

Reply:
-  kubernetes-sigs/jobset#407

Merged and part of v0.4.0

Reply:
Yeah but this is based on label and we decided to use field upstream

Reply:
The current state of the jobset fits our needs , if at some point  they decide to switch to field we can adapt. But for now I don't see that need.

Reply:
I agree that there is no need, but being a kubernetes-sig project, it should follow the same pattern as kubernetes.

So with the goal of standardizing and incentivizing other projects to follow suit, we can do the change in jobset.

Reply:
Yes, I think we want the clear path for other Cards to follow, so we prefer consistence. Second the same argument applies as for k8s job, that with label you may use it on old versions, and it may have unexpected consequences later on. With field we fail fast.

Reply:
There is an issue for this already https://github.com/kubernetes-sigs/jobset/issues/440

Reply:
I could do that also, but it should not block this PR. And we should realign when a new versin of jobset is release.

It will also help if we are sure the approach is final for Jobs.

Reply:
I think we should kick off the JobSet effort as soon as possible. If JobSet can support it quickly then my preference is to adapt this PR.

Reply:
I see no valid reason to hold this PR

Reply:
The reason is that we don't want to support for label is 0.7, because it would mean we need to maintain it in the future. If we merge this PR we need to be careful to remember to use field before the release.

Reply:
> The reason is that we don't want to support label in 0.7, because it would mean we need to maintain it in the future. If we merge this PR, in the current form, we need to be careful to remember to use field before the release.

Is JobSet going to support both at any point?

Reply:
There is a deprecation cost that needs to be paid for the label. I think we already have that at the JobSet level, but not yet at the Kueue level.

 I think it is preferred not to support label in Kueue at all, to avoid the cost for kueue.

Reply:
I think alpha features are mostly for experimentation so jobset is free to drop the support actually for the alpha level. If there is a user that urgently needs multikueue, which I doubt, they will be able to use Kueue 0.7 with main.

Reply:
@danielvegamyhre and I are discussing this https://github.com/kubernetes-sigs/jobset/issues/440#issuecomment-2015616035.

Ideally we would like to drop this ASAP as long as no users are using it.

Reply:
> I think alpha features are mostly for experimentation so jobset is free to drop the support (in the next release) actually for the alpha level label.

This is true but JobSet is a special case because all JobSet features are technically still alpha but several companies depend on it for their production workloads, making it more like v1/GA is that regard.

For this particular label though, I'm not aware of any users that actually use it, and it's only existed for 3 weeks, so perhaps we can just drop it in the April release.

Reply:
It sounds unlikely that anyone would use the managed-by label in a production environment. The only known use case is MultiKueue, and it's currently not depending on the label.

So, I think it would be ok just to drop the support for the label in the next release of jobset.

Reply:
I see, ok sounds good to me.

Reply:
@trasc please adapt this PR as the Jobset PR is merged: https://github.com/kubernetes-sigs/jobset/pull/487.

Reply:
We should wait for the change to be released, since we need to also install a compatible version of  jobser operator.

Reply:
I think we could for now install main version to move forward, then we just need to bump to 0.5 once JobSet is released. @danielvegamyhre any plans for the release of 0.5?

Reply:
Jobset 0.5 was released today!

Reply:
thanks